templates/common/_base/units/ovs-configuration.service.yaml

```diff
@@ -0,0 +1,19 @@
+name: ovs-configuration.service
+enabled: {{if eq .NetworkType "OVNKubernetes"}}true{{else}}false{{end}}
+contents: |
+  [Unit]
+  Description=Configures OVS with proper host networking configuration
+  # This service is used to move a physical NIC into OVS and reconfigure OVS to use the host IP
+  Requires=NetworkManager-wait-online.service openvswitch.service
+  After=NetworkManager-wait-online.service openvswitch.service node-valid-hostname.service
+  Before=network-online.target kubelet.service crio.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
```

Comment:
Do we need to set `RemainAfterExit=true` for this?

Also, what happens when we stop the openvswitch service? Are we supposed to deconfigure what this script configured?

Reply:
I don't think we need it. Are you suggesting having it so we can check its status? Or are you saying we should implement an exec stop as well?

If someone stops OVS they will lose network connectivity. If they are stopping it for some administrative reason on the host, then there's no reason to remove the config that this script has done in NM. The NM connections will remain configured, but we also leave the original connections configured in NM with a lower priority. So the admin could deactivate the OVS connection and bring up the old connection if he/she desired.

Reply:
I was thinking whatever we do to move the primary interface under OVS should be undone when OVS is stopped and moved back outside the ovs bridge as a net device of it's own on the stack.

Reply:
So in that sense, yes, implement a `ExecStop`.

Reply:
Hmm so in that case if OVS stops, we get a stop on this service, and then we nmcli down the OVS connection and bring up the old connection? I'm not sure of the implications there because I think NM will try to access ovsdb to delete the config (which won't be up anymore). I think we can address this as a follow up.

Reply:
Sure. Please have a github issue/BZ/JIRA to track this.

Reply:
Shouldn't a systemd directive be sufficient to take care of the ordering that is when OVS is stopped, this configuration service should get a chance to delete the config (while ovsdb is still up) and then set itself as stopped?